core/dbt/graph/selector.py

```diff
@@ -258,8 +258,21 @@ def get_selected(self, spec: SelectionSpec) -> Set[UniqueId]:
                 - selectors can filter the nodes after all of them have been
                   selected
         """
-        selected_nodes = self.select_nodes(spec)
+        selected_nodes, indirect_only = self.select_nodes(spec)
         filtered_nodes = self.filter_selection(selected_nodes)
+        
+        if indirect_only:
+            filtered_unused_nodes = self.filter_selection(indirect_only)
+            # log anything that didn't make the cut
+            if filtered_unused_nodes:
+                unused_node_names = []
+                for unique_id in filtered_unused_nodes:
+                    name = self.manifest.nodes[unique_id].name
+                    unused_node_names.append(name)
+                unused_nodes_str = ("\n  - ").join(unused_node_names)
+                msg = f"\nSome resources were excluded because at least one parent is missing:\n  - {unused_nodes_str}\nUse the --greedy flag to include them"
```

Comment:
This is the opposite to the phrasing you had in the sample code you wrote. I switched it because it was confusing to say "these were selected but they're not going to run", but I'm not super happy with how it reads now either.